src/CoreBundle/private/modules/CMSModulePageTreeRPC/CMSModulePageTreeRPC.class.php

```diff
@@ -99,23 +101,24 @@ public function getChildren()
      */
     public function getChildrenPortalCondition()
     {
-        $oUser = TCMSUser::GetActiveUser();
-        $sPortalList = $oUser->oAccessManager->user->portals->PortalList();
-        $query = 'SELECT * FROM `cms_portal`';
-        if (false !== $sPortalList) {
-            $query .= ' WHERE `id` NOT IN ('.$sPortalList.')';
+        /** @var SecurityHelperAccess $securityHelper */
+        $securityHelper = ServiceLocator::get(SecurityHelperAccess::class);
+        $portalIds = $securityHelper->getUser()?->getPortals();
+        $sPortalList = '';
+        if (null !== $portalIds && count($portalIds) >0) {
+            $sPortalList = implode(', ', array_map(fn(string $portalId) => $this->getDatabaseConnection()->quote($portalId), array_keys($portalIds)));
         }
-        $aPortalExcludeList = array();
-        $portalRes = MySqlLegacySupport::getInstance()->query($query);
-        while ($portal = MySqlLegacySupport::getInstance()->fetch_assoc($portalRes)) {
-            if (!empty($portal['main_node_tree'])) {
-                $aPortalExcludeList[] = MySqlLegacySupport::getInstance()->real_escape_string($portal['main_node_tree']);
-            }
+
+        $query = 'SELECT `main_node_tree` FROM `cms_portal`';
+        if ('' !== $sPortalList) {
+            $query .= ' WHERE `id` NOT IN ('.$sPortalList.')';
         }
+        $portalMainNodes = $this->getDatabaseConnection()->fetchAllAssociative($query);
+        $aPortalExcludeList = array_map(fn($row) => $this->getDatabaseConnection()->quote($row['main_node_tree']), $portalMainNodes);
 
         $sPortalCondition = '';
         if (count($aPortalExcludeList) > 0) {
-            $sPortalCondition .= " AND T.`id` NOT IN ('".implode("','", $aPortalExcludeList)."')";
+            $sPortalCondition .= " AND T.`id` NOT IN ('".implode(', ', $aPortalExcludeList)."')";
```

Comment:
Is it ok to drop the quotes from the query here?

Reply:
yes. see line 117. The entries are quoted. but there was a mistake. I needed to remove the `'` the follow and prefix the brackets